File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/logic/customdrops/recipient/DropTableRecipient.java
package io.github.arcaneplugins.levelledmobs.bukkit.logic.customdrops.recipient;

import io.github.arcaneplugins.levelledmobs.bukkit.logic.customdrops.CustomDrop;
import io.github.arcaneplugins.levelledmobs.bukkit.util.modal.impl.ModalEntityTypeSet;
import java.util.Collection;
import javax.annotation.Nonnull;
import org.jetbrains.annotations.NotNull;

public class DropTableRecipient extends CustomDropRecipient {

    private final String id;
    private final ModalEntityTypeSet applicableEntities;

    public DropTableRecipient(
        final @Nonnull String id,
        final @NotNull Collection<CustomDrop> drops,
        final float overallChance,
        final @NotNull Collection<String> overallPermissions,
        final ModalEntityTypeSet applicableEntities
    ) {
        super(drops, overallChance, overallPermissions);
        this.applicableEntities = applicableEntities;
        this.id = id;
    }

    @NotNull
    @Override
    public CustomDropRecipientType getRecipientType() {
        return CustomDropRecipientType.DROP_TABLE;
    }

    @Nonnull
    public ModalEntityTypeSet getApplicableEntityTypes() {
        return applicableEntities;
    }

    @Nonnull
    public String getId() {
        return id;
    }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/customdrops/recipient/DropTableRecipient.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.customdrops.recipient

import io.github.arcaneplugins.levelledmobs.bukkit.logic.customdrops.CustomDrop
import io.github.arcaneplugins.levelledmobs.bukkit.util.modal.impl.ModalEntityTypeSet

class DropTableRecipient(
    val id: String,
    drops: MutableList<CustomDrop>,
    overallChance: Float,
    overallPermissions: MutableList<String>,
    val applicableEntities: ModalEntityTypeSet
) : CustomDropRecipient(drops, overallChance, overallPermissions) {
    override fun getRecipientType(): CustomDropRecipientType {
        return CustomDropRecipientType.DROP_TABLE
    }
}